include/onnxruntime/core/graph/rewrite_rule.h

```diff
@@ -54,6 +69,63 @@ class GraphEditor {
     return graph_.Resolve();
   }
 
+  // TODO: The following functions should probably be moved to the Graph and Node classes.
+  // We also need to decide whether we keep the GraphEditor or we pass directly the Graph.
+  // If we keep it, it might be better to put it in a separate file.
+
+  /** Check whether the node has a single input and a single output. */
+  bool IsSingleInSingleOutNode(const Node& node) const {
```

Comment:
Are you expecting other callers, if not, move this as a static function in the implementation (.cc).

Reply:
I think it is a generic function that can be used by anyone. I would prefer to actually move it to the Graph class. Thoughts?

Reply:
it could be a generic helper function, but not an instance method of either graph or graph editor.

Reply:
Moved it to the new graph_utils.